operations/helm/charts/grafana-agent/values.yaml

```diff
@@ -9,12 +9,12 @@ fullnameOverride: null
 agent:
   # -- Mode to run Grafana Agent in. Can be "flow" or "static".
   mode: 'flow'
-
+  structuredConfig: {}
```

Comment:
Can you explain why we need both `structuredConfig` and `configMap.content`? I'm worried this adds too much extra complexity for understanding which values you need to set.

Reply:
I've found the structured config super useful when deploying things like loki - because it allows me to reference values that i've set in the yaml-structured way.

so lets say I set an override, and want to use it over and over again, i can do it this way

Reply:
I'm not sure I'm following. Even though the default value of `configMap.content` is a string, because we pass it through `toYaml`, can't you also give it a structured object?

Reply:
Ok, so I tried 

```yaml
agent:
  mode: static
  configMap:
    content:
      server:
        log_level: debug
``` 

and it doesn't work. But I would rather find a way to support `agent.configMap.content` to be either a string or an object rather than introducing a new field.

Reply:
So its more about 'the future' I think.  Currently this chart ships with no usable config for the agent - but I assume that will change in the future.  

Under that assumption this allows users to edit/override ANY key easily in a structured way - allowing a downstream user to inherit the 'sane defaults' of the chart, configuration, parameters, but easily provide a non-string value to change a key thing here-or-there.  

I'm not tied to this change personally - it is just what's been in place with [tempo](https://github.com/grafana/helm-charts/blob/main/charts/tempo-distributed/values.yaml#L74) and [loki](https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml#L281) and I've found that using it makes maintaining upgrades and configuration more pleasant for the end user relative to the alternative

Reply:
there are also just silly nice-to-have-things, like because its not an inline string, it looks prettier in the IDE or that you can use jq to edit/parse/validate easier but thats not really super important to most

Reply:
>But I would rather find a way to support agent.configMap.content to be either a string or an object rather than introducing a new field.

ok! let me back this out of THIS pr and ill throw up a separate one with that change